src/libcrun/utils.c

```diff
@@ -1849,48 +1849,62 @@ parse_json_file (yajl_val *out, const char *jsondata, struct parser_context *ctx
   return 0;
 }
 
+/* check that the specified path exists and it is executable.
+   Return
+   - 0 if the file is executable
+   - 1 if it exists but it is not executable
+   - 2 if it not a regular file
+   - -errno for any other generic error
+ */
 static int
 check_access (const char *path)
 {
   int ret;
   mode_t mode;
 
-#ifdef ANDROID
-  ret = access (path, X_OK);
+#ifdef HAVE_EACCESS
+#  define ACCESS(path, mode) (access (path, mode))
```

Comment:
I am not an experienced c dev but I think this should call eaccess(), otherwise there is no reason for the macro definition

Reply:
yes thanks.  I've screwed it up while testing the patch to check the access code path too, but I amended the wrong change.  I'll fix it